index.js

```diff
@@ -62,41 +62,33 @@ function createWarning (name, code, message, { unlimited = false } = {}) {
 
   code = code.toUpperCase()
 
-  return new WarningItem(name, code, message, unlimited)
-}
+  let warning
 
-/**
- * Represents a warning item with details.
- * @class
- * @memberof processWarning
- * @param {string} name - The name of the warning.
- * @param {string} code - The code associated with the warning.
- * @param {string} message - The warning message.
- * @param {boolean} unlimited - If true, allows unlimited emissions of the warning.
- */
-class WarningItem {
-  constructor (name, code, message, unlimited) {
-    this.name = name
-    this.code = code
-    this.message = message
-    this.unlimited = unlimited
-    this.emitted = false
-  }
+  const warningContainer = unlimited
+    ? {
```

Comment:
For what it's worth, especially considering the large object literal definitions (multiple lines), I would do:

```js
let warningContainer = { [name]: function () {} }
if (unlimited === true) {
  warningContainer[name] = /* stuff */
} else {
  /* other stuff */
}
```

Ternaries are difficult to read and should be reserved for much simpler cases, in my opinion.

Reply:
It would be

```js
  let warningContainer = {
    [name]: function (a, b, c) {
      if (warning.emitted === true && warning.unlimited !== true) {
        return
      }
      warning.emitted = true

      process.emitWarning(warning.format(a, b, c), warning.name, warning.code)
    }
  }
  if (unlimited) {
    warningContainer = {
      [name]: function (a, b, c) {
        warning.emitted = true

        process.emitWarning(warning.format(a, b, c), warning.name, warning.code)
      }
    }
  }
```

Do you want it this way?

Reply:
I think it is easier to read, yes. And with your suggestion, you don't have to redefine the whole object, just replace the method.